altsrc/map_input_source.go

```diff
@@ -127,19 +128,46 @@ func (fsm *MapInputSource) Float64(name string) (float64, error) {
 	return 0, nil
 }
 
+func castToInt64(v interface{}) (int64, bool) {
+	int64Value := int64(0)
+	isType := false
+	if v == nil {
+		return int64Value, true
+	}
+
+	// There are only four cases(int, int64, uint64, float64) when parsing the integer in yaml.v3 pkg
+	// But the last case, float64, is an error case so that ignored
+	vType := reflect.TypeOf(v).Kind()
```

Comment:
does reflect need to be used ? You cant just do 

```
switch v.(type) {
case uint64:
   ...
}
```
?

Reply:
Even simpler can use:

```go
// quick example
switch val := v.(type) {
    case int:
        int64Value = int64(val)
    case int64:
        int64Value = val
}
```

Reply:
resolved it. fb6fcd2